src/services/TestResultsService.ts

```diff
@@ -170,14 +170,22 @@ export class TestResultsService {
     }
     payload.testTypes = testTypes;
     return this.testResultsDAO.getBySystemNumber(systemNumber)
-        .then((result) => {
+        .then(async (result) => {
           const response: ITestResultData = {Count: result.Count, Items: result.Items};
           const testResults = this.checkTestResults(response);
           const oldTestResult = this.getTestResultToArchive(testResults, payload.testResultId);
           oldTestResult.testVersion = TEST_VERSION.ARCHIVED;
           const newTestResult: ITestResult = cloneDeep(oldTestResult);
           newTestResult.testVersion = TEST_VERSION.CURRENT;
           mergeWith(newTestResult, payload, this.arrayCustomizer);
+          if (this.shouldGenerateNewTestCodeRe(oldTestResult, newTestResult)) {
+            await this.getTestTypesWithTestCodesAndClassification(newTestResult.testTypes as any[],
+              newTestResult.vehicleType, newTestResult.vehicleSize, newTestResult.vehicleConfiguration,
+              newTestResult.noOfAxles, newTestResult.euVehicleCategory, newTestResult.vehicleClass.code,
+              newTestResult.vehicleSubclass ? newTestResult.vehicleSubclass[0] : undefined,
```

Comment:
change the condition to `newTestResult.vehicleSubclass && newTestResult.vehicleSubclass.length` is better approach for an array. Also move it one line above outside the function to increase the readability.